rpc/transaction.go

```diff
@@ -118,20 +118,24 @@ func (provider *Provider) WaitForTransaction(ctx context.Context, transactionHas
 			}
 			switch r := receipt.(type) {
 			case DeclareTransactionReceipt:
-				if r.Status.IsTransactionFinal() {
-					return r.Status, nil
+				if r.ExecutionStatus == TxnExecutionStatusSUCCEEDED {
+					return r.ExecutionStatus, nil
+				}
+			case DeployTransactionReceipt:
+				if r.ExecutionStatus == TxnExecutionStatusSUCCEEDED {
+					return r.ExecutionStatus, nil
 				}
 			case DeployAccountTransactionReceipt:
-				if r.Status.IsTransactionFinal() {
-					return r.Status, nil
+				if r.ExecutionStatus == TxnExecutionStatusSUCCEEDED {
+					return r.ExecutionStatus, nil
 				}
 			case InvokeTransactionReceipt:
-				if r.Status.IsTransactionFinal() {
-					return r.Status, nil
+				if r.ExecutionStatus == TxnExecutionStatusSUCCEEDED {
+					return r.ExecutionStatus, nil
 				}
 			case L1HandlerTransactionReceipt:
-				if r.Status.IsTransactionFinal() {
-					return r.Status, nil
+				if r.ExecutionStatus == TxnExecutionStatusSUCCEEDED {
+					return r.ExecutionStatus, nil
```

Comment:
Maybe add execution status to the interface and check for failure before the switch? i.e.
```
if receipt.ExecutionStatus() != ...SUCCEEDED {
    return ...
 }
 
 switch ...
 ```

Reply:
I added the GetExecutionStatus() method to the interface. I also think we can just return the execution status, without having to type cast in the switch statement